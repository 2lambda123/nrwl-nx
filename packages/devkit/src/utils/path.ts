import * as path from 'path';

function removeWindowsDriveLetter(osSpecificPath: string): string {
  return osSpecificPath.replace(/^[A-Z]:/, '');
}

/**
 * Coverts an os specific path to a unix style path
 */
export function normalizePath(osSpecificPath: string): string {
  /*
  why not use https://nodejs.org/docs/latest-v14.x/api/path.html#path_path_normalize_path (available since v0.1.23)?
   */
  return removeWindowsDriveLetter(osSpecificPath).split('\\').join('/');
}

/**
 * Normalized path fragments and joins them
 */
export function joinPathFragments(...fragments: string[]): string {
  return normalizePath(path.join(...fragments));
}
