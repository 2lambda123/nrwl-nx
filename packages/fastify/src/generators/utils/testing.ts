import { Tree } from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';

export function createTreeWithFastifyApplication(appName: string): Tree {
  const tree = createTreeWithEmptyWorkspace();
  tree.write(
    'workspace.json',
    String.raw`
      {
        "projects": {
          "${appName}": {
            "root": "apps/api",
            "sourceRoot": "apps/api/src",
            "projectType": "application",
            "prefix": "api",
            "targets":{}
          }
        }
      }
    `
  );

  return tree;
}
