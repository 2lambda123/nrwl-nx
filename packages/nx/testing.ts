export {cold, hot} from 'jasmine-marbles';
export {readAll} from './src/testing-utils';
