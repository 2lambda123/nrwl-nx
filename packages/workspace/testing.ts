export { MockBuilderContext } from './src/utils/testing-utils';
