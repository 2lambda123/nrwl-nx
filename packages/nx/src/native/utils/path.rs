use crate::native::utils::normalize_trait::Normalize;
use std::path::{Path, PathBuf};

impl Normalize for Path {
    fn to_normalized_string(&self) -> String {
        normalize_nx_path(self)
    }
}

impl Normalize for PathBuf {
    fn to_normalized_string(&self) -> String {
        normalize_nx_path(self)
    }
}

fn normalize_nx_path<P>(path: P) -> String
where
    P: AsRef<Path>,
{
    if path.as_ref() == Path::new("") {
        return ".".into();
    }

    // convert back-slashes in Windows paths, since the js expects only forward-slash path separators
    if cfg!(windows) {
        path.as_ref().display().to_string().replace('\\', "/")
    } else {
        path.as_ref().display().to_string()
    }
}
