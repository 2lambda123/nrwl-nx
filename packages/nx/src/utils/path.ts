import * as path from 'path';

function removeWindowsDriveLetter(osSpecificPath: string): string {
  return osSpecificPath.replace(/^[A-Z]:/, '');
}

/**
 * Coverts an os specific path to a unix style path
 */
export function normalizePath(osSpecificPath: string): string {
  return removeWindowsDriveLetter(osSpecificPath).split('\\').join('/');
}

/**
 * Normalized path fragments and joins them
 */
export function joinPathFragments(...fragments: string[]): string {
  return normalizePath(path.join(...fragments));
}

/**
 * Detect workspace scope from the package.json name
 * @param packageName
 * @returns
 */
export function detectWorkspaceScope(packageName: string): string {
  if (!packageName) return '';

  if (packageName.startsWith('@')) {
    return packageName.substring(1).split('/')[0];
  }

  if (packageName.includes('/')) {
    return packageName.split('/')[0];
  }

  return '';
}

/**
 * Prefixes project name with npm scope
 */
export function getImportPath(
  npmScope: string,
  projectDirectory: string
): string {
  return npmScope ? `@${npmScope}/${projectDirectory}` : projectDirectory;
}
