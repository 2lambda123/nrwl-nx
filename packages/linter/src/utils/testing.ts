import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import { join } from 'path';
import { Tree } from '@angular-devkit/schematics';

const migrationTestRunner = new SchematicTestRunner(
  '@nrwl/linter-migrations',
  join(__dirname, '../../migrations.json')
);

const testRunner = new SchematicTestRunner(
  '@nrwl/linter',
  join(__dirname, '../../collection.json')
);

export function runMigration<T = any>(
  migrationName: string,
  options: T,
  tree: Tree
) {
  return migrationTestRunner
    .runSchematicAsync<T>(migrationName, options, tree)
    .toPromise();
}

export function runSchematic<SchemaOptions = any>(
  schematicName: string,
  options: SchemaOptions,
  tree: Tree
) {
  return testRunner.runSchematicAsync(schematicName, options, tree).toPromise();
}
