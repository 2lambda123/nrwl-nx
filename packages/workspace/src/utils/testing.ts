import { join } from 'path';
import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import type { Rule, Tree } from '@angular-devkit/schematics';

const testRunner = new SchematicTestRunner(
  '@nrwl/workspace',
  join(__dirname, '../../generators.json')
);

testRunner.registerCollection(
  '@nrwl/jest',
  join(__dirname, '../../../jest/generators.json')
);

testRunner.registerCollection(
  '@nrwl/cypress',
  join(__dirname, '../../../cypress/generators.json')
);

testRunner.registerCollection(
  '@nrwl/express',
  join(__dirname, '../../../express/generators.json')
);

testRunner.registerCollection(
  '@nrwl/react',
  join(__dirname, '../../../react/generators.json')
);

testRunner.registerCollection(
  '@nrwl/angular',
  join(__dirname, '../../../angular/generators.json')
);

testRunner.registerCollection(
  '@nrwl/next',
  join(__dirname, '../../../next/generators.json')
);

testRunner.registerCollection(
  '@nrwl/node',
  join(__dirname, '../../../node/generators.json')
);

testRunner.registerCollection(
  '@nrwl/nest',
  join(__dirname, '../../../nest/generators.json')
);

testRunner.registerCollection(
  '@nrwl/web',
  join(__dirname, '../../../web/generators.json')
);

/**
 * @deprecated This will be removed in v17. Prefer writing Nx Generators with @nrwl/devkit. Generators can be tested by simply calling them in it().
 */
export function runSchematic<T extends object = any>(
  schematicName: string,
  options: T,
  tree: Tree
) {
  return testRunner.runSchematicAsync(schematicName, options, tree).toPromise();
}

/**
 * @deprecated This will be removed in v17. Prefer writing Nx Generators with @nrwl/devkit. Generators can be tested by simply calling them in it().
 */
export function callRule(rule: Rule, tree: Tree) {
  return testRunner.callRule(rule, tree).toPromise();
}
