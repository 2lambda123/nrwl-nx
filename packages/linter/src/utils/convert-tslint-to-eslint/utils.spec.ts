import { addProjectConfiguration, Tree } from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import type { Linter } from 'eslint';
import { join } from 'path';
import {
  convertTSLintDisableCommentsForProject,
  deduplicateOverrides,
} from './utils';

describe('deduplicateOverrides()', () => {
  it('should deduplicate overrides with identical values for "files"', () => {
    const initialOverrides: Linter.Config['overrides'] = [
      {
        files: ['*.ts'],
        env: {
          foo: true,
        },
        rules: {
          bar: 'error',
        },
      },
      {
        files: ['*.html'],
        rules: {},
      },
      {
        files: '*.ts',
        plugins: ['wat'],
        parserOptions: {
          qux: false,
        },
        rules: {
          bar: 'warn',
          baz: 'error',
        },
      },
      {
        files: ['*.ts'],
        extends: ['something'],
      },
    ];
    expect(deduplicateOverrides(initialOverrides)).toEqual([
      {
        files: ['*.ts'],
        env: {
          foo: true,
        },
        plugins: ['wat'],
        extends: ['something'],
        parserOptions: {
          qux: false,
        },
        rules: {
          bar: 'warn',
          baz: 'error',
        },
      },
      {
        files: ['*.html'],
        rules: {},
      },
    ]);
  });
});

describe('convertTSLintDisableCommentsForProject', () => {
  let tree: Tree;
  const projectName = 'foo';
  const projectRoot = `apps/${projectName}`;

  beforeEach(async () => {
    tree = createTreeWithEmptyWorkspace();
    addProjectConfiguration(tree, projectName, {
      root: projectRoot,
      projectType: 'application',
      targets: {
        lint: {
          executor: '@angular-devkit/build-angular:tslint',
          options: {
            exclude: ['**/node_modules/**', `!${projectRoot}/**/*`],
            tsConfig: [`${projectRoot}/tsconfig.app.json`],
          },
        },
      },
    });
    tree.write(
      join(projectRoot, 'top-level-file.ts'),
      `
      // tslint:disable
      eval('');
    `
    );
    tree.write(
      join(projectRoot, 'single-level-nested/file.ts'),
      `
      // tslint:disable-next-line
      eval('');
    `
    );
    // specific rule, and multi-line-comment style
    tree.write(
      join(projectRoot, 'multi-level/nested/file.ts'),
      `
      /* tslint:disable:quotemark */
      eval('');
    `
    );
  });
  it('should replace tslint:disable comments with their ESLint equivalents in .ts files for the given project', () => {
    convertTSLintDisableCommentsForProject(tree, projectName);

    expect(
      tree.read(join(projectRoot, 'top-level-file.ts')).toString()
    ).toMatchSnapshot();
    expect(
      tree.read(join(projectRoot, 'single-level-nested/file.ts')).toString()
    ).toMatchSnapshot();
    expect(
      tree.read(join(projectRoot, 'multi-level/nested/file.ts')).toString()
    ).toMatchSnapshot();
  });
});
