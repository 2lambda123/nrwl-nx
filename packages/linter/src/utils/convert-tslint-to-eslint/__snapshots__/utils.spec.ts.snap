// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertTSLintDisableCommentsForProject should replace tslint:disable comments with their ESLint equivalents in .ts files for the given project 1`] = `
"
      /* eslint-disable */
      eval('');
    "
`;

exports[`convertTSLintDisableCommentsForProject should replace tslint:disable comments with their ESLint equivalents in .ts files for the given project 2`] = `
"
      // eslint-disable-next-line
      eval('');
    "
`;

exports[`convertTSLintDisableCommentsForProject should replace tslint:disable comments with their ESLint equivalents in .ts files for the given project 3`] = `
"
      /* eslint-disable @typescript-eslint/quotes */
      eval('');
    "
`;
